module/VuFind/src/VuFind/Db/Service/AccessTokenService.php

```diff
@@ -96,7 +105,11 @@ public function getByIdAndType(
      */
     public function storeNonce(int $userId, ?string $nonce): void
     {
-        $this->accessTokenTable->storeNonce($userId, $nonce);
+        $dql = 'UPDATE ' . $this->getEntityClass(AccessToken::class) . ' at '
+                . 'SET at.data = :nonce WHERE at.user = :user';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters(['nonce' => $nonce, 'user' => $userId]);
```

Comment:
Probably worth using `$this->entityManager->getReference(UserEntityInterface::class, $userId)` here -- will increase flexibility if we want to broaden the type of $userId in future.